andela/jubilee-backend
# #167733657 User Role Settings

#### What does this PR do?
Assigns Roles to users, It also checks user permissions before accessing specific routes, allows a Super Administrator to change a user's role.

#### Description of Task to be completed?
- Create the database models, migrations and table associations
- Create middleware for user role checks
- Create endpoints for user role status change

#### How should this be manually tested?
- Clone the repo and cd into app root
-  Setup necessary environment variables for testing environment (check .sample.env)
- Run docker-compose up --build to build docker image or run locally using `npm run start:dev`
- Run `npm run pretest` to migrate and seed database 
-  run `npm run test` to run all tests
- On postman, run `localhost:3000/api/auth/signup/user` with user json parameters, eg:
```
{
	"firstName": "kunle",
	"lastName": "Mnta",
	"email": "kunle@montan.com",
	"password": "tmobnvarq.ss66u",
	"gender": "male",
	"street": "Abage",
	"city": "Lagos",
	"state": "Imo",
	"country": "Nigeria",
	"birthdate": "2019-08-20",
	"phoneNumber": "(+234) 090-3472-3889",
	"companyName": "Jubilee"
}
```
- On postman, run `localhost:3000/api/users/role` for role reassignment (make sure you have a superUser token in cookie).
#### Any background context you want to provide?
none

### screenshots (if appropriate)

<img width="1660" alt="Screenshot 2019-08-30 at 2 50 29 PM" src="https://user-images.githubusercontent.com/37340699/64065853-d6620c80-cc0a-11e9-9b70-adc126ced21b.png">

<img width="831" alt="Screenshot 2019-08-30 at 10 32 49 AM" src="https://user-images.githubusercontent.com/37340699/64065856-dfeb7480-cc0a-11e9-9a05-1d1ec6990cb3.png">


#### What are the relevant pivotal tracker stories?
[#1167733657](https://www.pivotaltracker.com/story/show/167733657)
